snap/seed_yaml.go

```diff
@@ -89,6 +92,20 @@ func ReadSeedYaml(fn string) (*Seed, error) {
 		if strings.Contains(sn.File, "/") {
 			return nil, fmt.Errorf("%s: %q must be a filename, not a path", errPrefix, sn.File)
 		}
+
+		// make sure names, snap IDs and file names are unique
+		if sn.SnapID != "" && seenSnapIDs[sn.SnapID] {
+			return nil, fmt.Errorf("%s: snap-id %q must be unique", errPrefix, sn.SnapID)
+		}
+		seenSnapIDs[sn.SnapID] = true
+		if seenNames[sn.Name] {
+			return nil, fmt.Errorf("%s: snap name %q must be unique", errPrefix, sn.Name)
+		}
+		seenNames[sn.Name] = true
+		if seenFiles[sn.File] {
```

Comment:
Silly question - will this prevent us from installing instances in seed.yaml? I.e. something like:
```
snaps:
 - name: foo
   file: foo.snap
 - name: foo_instance
   file: foo.snap
```
or is this not supported already anyway?

Reply:
Very good point, I missed that! I don't think we prevent this (or should prevent). I've reverted this change (as well checking for unique snap ids) for now as they need to be smarter if we want this check - a possible followup. So for now this PR just checks unique names.

Reply:
we shouldn't support installing instances ATM